packages/react/tsconfig.json

```diff
@@ -17,6 +17,6 @@
     "module": "esnext",
     "target": "es2015"
   },
-  "exclude": ["node_modules", "src/**/*.test.ts", "src/**/*.test.tsx"],
+  "exclude": ["node_modules"],
```

Comment:
I'm not sure why we excluded tests from our tsconfig. But I guess it's fine not to exclude them.

Reply:
Last time I was trying to remove them but then I found `.d.ts` will also be generated for test files and shipped along with other code. Can you verify that?

Reply:
That was a miss on my end, going to put them back. 

That being said, it did allow me to catch an issue where one of the `AmplifyProvider` unit tests had an import from `@aws-amplify/ui-react`, which makes me think that there is probably value in adding a `tsconfig.dev.json` to run our type checking against in dev, but that should be a separate task

Reply:
Does ESslint always reference the settings in `tsconfig.json`? Is it able to override the exclude in its own config file?

Reply:
Type checking and eslint are tangental to each other, but for TS related rules we need to point eslint towards a `tsconfig` file. The `excludes` property is specific for type compilation in TS